SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -190,25 +194,77 @@ private Task ProcessPendingErrorsAsync()
                         _unprocessedManagedErrorLogs.Add(log.Id, log);
                     }
                 }
-                SendCrashReportsOrAwaitUserConfirmation();
-            }).ContinueWith((_) => ProcessPendingErrorsTask = null);
+                await SendCrashReportsOrAwaitUserConfirmationAsync();
+            });
         }
 
-        private void SendCrashReportsOrAwaitUserConfirmation()
+        private Task SendCrashReportsOrAwaitUserConfirmationAsync()
         {
-            HandleUserConfirmation();
+            return HandleUserConfirmationAsync();
         }
 
-        private void HandleUserConfirmation()
+        private Task HandleUserConfirmationAsync()
         {
             // Send every pending log.
             var keys = _unprocessedManagedErrorLogs.Keys.ToList();
+            var tasks = new List<Task>();
             foreach (var key in keys)
             {
-                Channel.EnqueueAsync(_unprocessedManagedErrorLogs[key]);
+                var log = _unprocessedManagedErrorLogs[key];
+                tasks.Add(Channel.EnqueueAsync(log));
                 _unprocessedManagedErrorLogs.Remove(key);
                 ErrorLogHelper.RemoveStoredErrorLogFile(key);
+                
+                // Save callback into a local variable to avoid a race condition.
+                var errorAttachmentsCallback = GetErrorAttachments;
+                if (errorAttachmentsCallback != null)
+                {
+                    var errorReport = new ErrorReport(log, null);
+
+                    // This must never called while a lock is held.
+                    var attachments = errorAttachmentsCallback.Invoke(errorReport);
+                    tasks.Add(SendErrorAttachmentsAsync(log.Id, attachments));
```

Comment:
Checking empty attachments here before creating tasks would help to reduce producing the number of Tasks.